### PR TITLE
JAVACLIENT-150: Add Annotation API

### PR DESCRIPTION
**Pull request by kevinleturc**

### CODE REVIEW

**Comment by driverpt** [comment]
👍

File: nuxeo-java-client-test/src/test/java/org/nuxeo/client/ITRepository.java [hunk 1]
```diff
@@ -695,4 +694,37 @@ private String buildErrorMsgForDate(String key, Class<?> valueType) {
                 key, valueType.getTypeName());
     }
 
+    @Test
+    public void itCanManageAnnotations() {
+        assumeTrue("itCanManageAnnotations works only since Nuxeo 10.2",
+                // TODO change the version to LTS
```
Comment (jcarsique) [inline nuxeo-java-client-test/src/test/java/org/nuxeo/client/ITRepository.java]: remove the TODO or associate with an open JIRA ticket

File: nuxeo-java-client/src/main/java/org/nuxeo/client/objects/annotation/Annotation.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package org.nuxeo.client.objects.annotation;
```
Comment (jcarsique) [inline nuxeo-java-client/src/main/java/org/nuxeo/client/objects/annotation/Annotation.java]: missing file header

File: nuxeo-java-client/src/main/java/org/nuxeo/client/objects/annotation/Annotations.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package org.nuxeo.client.objects.annotation;
```
Comment (jcarsique) [inline nuxeo-java-client/src/main/java/org/nuxeo/client/objects/annotation/Annotations.java]: file header

File: nuxeo-java-client-test/src/test/java/org/nuxeo/client/ITRepository.java (hunk 1 above)
Comment (kevinleturc) [reply nuxeo-java-client-test/src/test/java/org/nuxeo/client/ITRepository.java]: I won't, for each LTS release we need to do manual tasks on this client for tests, constants and this kind of stuff.

File: nuxeo-java-client/src/main/java/org/nuxeo/client/objects/annotation/Annotation.java (hunk 2 above)
Comment (kevinleturc) [reply nuxeo-java-client/src/main/java/org/nuxeo/client/objects/annotation/Annotation.java]: thx

File: nuxeo-java-client/src/main/java/org/nuxeo/client/objects/annotation/Annotations.java (hunk 3 above)
Comment (kevinleturc) [reply nuxeo-java-client/src/main/java/org/nuxeo/client/objects/annotation/Annotations.java]: thx

**Comment by kevinleturc** [comment]
There's currently no way to address the documentation for server version. There's ongoing work for documentation, this aspects will be reworked.

**Pull request merged**